packages/manager/apps/zimbra/src/hooks/useMailingList.ts

```diff
@@ -1,27 +1,36 @@
-import { useQuery } from '@tanstack/react-query';
+import {
+  useQuery,
+  UseQueryOptions,
+  UseQueryResult,
+} from '@tanstack/react-query';
 import { usePlatform } from '@/hooks';
 import {
   getZimbraPlatformMailingListDetails,
   getZimbraPlatformMailingListDetailsQueryKey,
+  MailingListType,
 } from '@/api/mailinglist';
 
-export const useMailingList = (mailingListId: string, noCache?: boolean) => {
+type UseMailingListParams = Omit<
+  UseQueryOptions,
+  'queryKey' | 'queryFn' | 'select'
+> & {
+  mailingListId?: string;
+};
+
+export const useMailingList = (props: UseMailingListParams = {}) => {
   const { platformId } = usePlatform();
-  const { data, isLoading, isError, error } = useQuery({
+  const { mailingListId, ...options } = props;
+  return useQuery({
+    ...options,
     queryKey: getZimbraPlatformMailingListDetailsQueryKey(
       platformId,
       mailingListId,
     ),
     queryFn: () =>
       getZimbraPlatformMailingListDetails(platformId, mailingListId),
-    enabled: !!mailingListId && !!platformId,
-    gcTime: noCache ? 0 : 5000,
-  });
-
-  return {
-    isLoading,
-    isError,
-    error,
-    data,
-  };
+    enabled:
+      (typeof options.enabled !== 'undefined' ? options.enabled : true) &&
```

Comment:
=== 'boolean' or anything else than boolean ( !== 'undefined' ) ?

Reply:
it just means that if `enabled` wasn't passed in the options we consider it true